esx_service/vmdk_ops_test.py

```diff
@@ -1201,6 +1175,211 @@ def test_vmdkops_for_default_datastore(self):
+
+        # delete should fail because policy is in use
+        self.assertNotEqual(None, vsan_policy.delete(self.name))
+
+        # Setting an identical policy returns an error msg
+        self.assertNotEqual(None, vsan_policy.update('good',
+                                    self.orig_policy_content))
+
+        backup_policy_file = vsan_policy.backup_policy_filename(self.name)
+        #Ensure there is no backup policy file
+        self.assertFalse(os.path.isfile(backup_policy_file))
+
+        # Fail to update because of a bad policy, and ensure there is no backup
+        self.assertNotEqual(None, vsan_policy.update('good', 'blah'))
+        self.assertFalse(os.path.isfile(backup_policy_file))
+
+        # try to delete the volume again, which should succeed
+        # run remove command
+        opts = {}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path,
+                                auth.CMD_REMOVE, self.default_tenant_vol3_name, opts)
+        self.assertEqual(None, error_info)
+
+        self.assertEqual(None, vsan_policy.delete(self.name))
+
+    @unittest.skipIf(not vsan_info.get_vsan_datastore(),
+                    "VSAN is not found - skipping vsan_info tests")
+    def testPolicyUsageCount(self):
+        self.orig_policy_content = ('(("proportionalCapacity" i0) '
+                                     '("hostFailuresToTolerate" i0))')
+
+        self.name = 'good'
+        vsan_policy.create(self.name, self.orig_policy_content)
+
+        vm1_uuid = vmdk_utils.get_vm_uuid_by_name(self.vm1_name)
+
+        # run create command
+        opts={'vsan-policy-name': 'good', u'fstype': u'ext4'}
+
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path,
+                                auth.CMD_CREATE, self.default_tenant_vol3_name, opts)
+        self.assertEqual(None, error_info)
+
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path,
+                                auth.CMD_CREATE, self.default_tenant_vol4_name, opts)
+        self.assertEqual(None, error_info)
+
+        associated_vmdk_count = 0
+        for v in vsan_policy.list_volumes_and_policies():
+            if v['policy'] == "good":
+                associated_vmdk_count = associated_vmdk_count + 1
+
+        self.assertEqual(associated_vmdk_count, 2,
+                        "volumes-policy association listing failed")
+
+        # try to delete the volume again, which should succeed
+        opts = {}
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path,
+                                auth.CMD_REMOVE, self.default_tenant_vol3_name, opts)
+        self.assertEqual(None, error_info)
+
+        error_info = vmdk_ops.executeRequest(vm1_uuid, self.vm1_name, self.vm1_config_path,
+                                auth.CMD_REMOVE, self.default_tenant_vol4_name, opts)
+        self.assertEqual(None, error_info)
+
```

Comment:
can you add similar check after removing volumes .. usage count should drop to 0 or ```vmdkops_admin.py policy ls```?